gcs/src/test/java/com/google/cloud/hadoop/fs/gcs/GoogleHadoopFileSystemTest.java

```diff
@@ -200,6 +201,25 @@ public void testGetDefaultPortIndicatesPortsAreNotUsed() throws Exception {
     assertThat(ghfs.getDefaultPort()).isEqualTo(-1);
   }
 
+  @Test
+  public void read_non_existing_file_throws_exception() throws Exception {
+    URI path = new URI("gs://test-non-existent/read-throws-exception");
+    Configuration configuration = new Configuration();
+
+    // try to read a non-existing file
+    configuration.setBoolean("fs.gs.inputstream.fast.fail.on.not.found.enable", false);
+    GoogleHadoopFileSystem ghfs =
+        GoogleHadoopFileSystemIntegrationHelper.createGhfs(path, configuration);
```

Comment:
This will try creating an actual connection with gcs, which would fail eventually and hence you are seeing `IOException` and test passes. But I believe that's not the intent of this unit test. In this test suite we create InMemory ghfs, try using that.

One suggestion would be to create an inMemeory file of size 1B and try reading from offset 100 this should fail with an exception.